File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        numFreqMap = {}
        for num in nums:
            numFreqMap[num] = numFreqMap.get(num, 0) + 1
        heap = []
        for num in numFreqMap.keys():
            heapq.heappush(heap, (numFreqMap[num], num))
            if len(heap) > k:
                heapq.heappop(heap)
        res = []
        for i in range(k):
            res.append(heapq.heappop(heap)[1])
        return res

        

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int farthestJump = 0;
        for (int i = 0; i < n; i++) {
            if (farthestJump < i) {
                return false;
            }
            farthestJump = max(farthestJump, i + nums[i]);
        }
        return true;
    }
};
